app/controllers/admin/dfc_product_imports_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require "private_address_check"
+require "private_address_check/tcpsocket_ext"
+
+module Admin
+  class DfcProductImportsController < Spree::Admin::BaseController
+    # Define model class for `can?` permissions:
+    def model_class
+      self.class
+    end
+
+    def index
+      # The plan:
+      #
+      # * Fetch DFC catalog as JSON from URL.
+      enterprise = OpenFoodNetwork::Permissions.new(spree_current_user)
+        .managed_product_enterprises.is_primary_producer
+        .find(params.require(:enterprise_id))
+
+      catalog_url = params.require(:catalog_url)
+
+      json_catalog = DfcRequest.new(spree_current_user).get(catalog_url)
```

Comment:
I am wondering if we need to rescue network related error so we can present a message to the user, and additional logging for debug purposes ?

Reply:
Yes, that would be great. But I think that we have to get the prototype working first before we have time to deal with these scenarios. But feel free to go ahead if you find some spare time.

Reply:
Fair enough! I do not know of this "spare time" you are talking about :laughing: